awx/ui_next/src/screens/User/shared/UserForm.jsx

```diff
@@ -68,11 +68,9 @@ function UserFormFields({ user, i18n }) {
         name="username"
         type="text"
         validate={
-          !ldapUser && externalAccount === null
-            ? required(null, i18n)
-            : () => undefined
+          !ldapUser && !externalAccount ? required(null, i18n) : () => undefined
         }
-        isRequired={!ldapUser && externalAccount === null}
+        isRequired={!ldapUser && !externalAccount}
```

Comment:
Common JavaScript surprise: If `externalAccount` is `{}`, then `!externalAccount` will be _false_. 

![Screenshot from 2020-10-27 17-01-41](https://user-images.githubusercontent.com/9753817/97362157-f5901480-1876-11eb-9811-e3b6dd4e1ac9.png)

Would that be the expected behavior here? The reason I ask is because in the PR summary there is a mention of `{}` being used as the default user value.

Reply:
It was the initial value of user `{}`, then `externalAccount` was undefined. I will add more unit-tests around this. Thanks for your review.

Reply:
@jakemcdermott, I added more unit-tests.